hack/lib/log.sh

```diff
@@ -34,6 +34,7 @@ function os::log::clean_up_logger() {
     set +o errexit
 
     if jobs -pr | grep -q "${LOGGER_PID}"; then
+        pgrep -P "${LOGGER_PID}" | xargs kill -SIGTERM &> /dev/null
```

Comment:
As mentioned in @deads2k PR, this is unnecessary. Killing `sar` with `SIGTERM` will propagate the signal to `sadc`. A better improvement to this code would be to remove the first `jobs` check, leaving us with:

```sh
kill -SIGTERM "${LOGGER_PID}"
# give logger ten seconds to gracefully exit before killing it
for (( i = 0; i < 10; i++ )); do
    if ! jobs -pr | grep -q "${LOGGER_PID}"; then
        # the logger has shutdown, we don't need to wait on it any longer
        break
    fi
done

if jobs -pr | grep -q "${LOGGER_PID}"; then
    # the logger has not shutdown, so kill it and it's child
    kill -SIGKILL "$( ps --ppid="${LOGGER_PID}" --format=pid --no-headers )"
    kill -SIGKILL "${LOGGER_PID}"
fi
```

 The first `jobs` check happened as originally the `LOGGER_PID` was incorrectly set to track the function, so I do not think we gain anything from the check any more.